### PR TITLE
fix: use require instead of t.Fatal(err) in tests/e2e package

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

This uses testify instead of testing for t.Fatal calls

Related to #18972

### CODE REVIEW

File: tests/e2e/ctl_v3_auth_test.go [hunk 1]
```diff
@@ -103,101 +103,77 @@ func authTestMemberUpdate(cx ctlCtx) {
...
-	require.NoError(cx.t, err)
-	err = ctlV3User(cx, []string{"grant-role", "example.com", "test-role"}, "Role test-role is granted to user example.com", nil)
-	require.NoError(cx.t, err)
+	require.NoError(cx.t, ctlV3User(cx, []string{"add", "example.com", "--interactive=false"}, "User example.com created", []string{""}))
+	require.NoError(cx.t, e2e.SpawnWithExpectWithEnv(append(cx.PrefixArgs(), "role", "add", "test-role"), cx.envMap, expect.ExpectedResponse{Value: "Role test-role created"}))
+	require.NoError(cx.t, ctlV3User(cx, []string{"grant-role", "example.com", "test-role"}, "Role test-role is granted to user example.com", nil))
 
 	// grant a new key
-	err = ctlV3RoleGrantPermission(cx, "test-role", grantingPerm{true, true, "hoo", "", false})
-	require.NoError(cx.t, err)
```
Comment (ahrtr) [inline tests/e2e/ctl_v3_auth_test.go:119]: question: did linter complain this?

**Comment by ahrtr** [comment]
This PR is just mechanical changes, but we love small PRs. I will let this PR in,  but please try to submit small PR , i.e. < 100 lines of code change as much as you can next time. Thanks.

File: tests/e2e/http_health_check_test.go [hunk 2]
```diff
@@ -172,33 +172,31 @@ func TestHTTPHealthHandler(t *testing.T) {
...
-		{
-			url:                    "/livez",
-			expectedStatusCode:     http.StatusOK,
-			expectedRespSubStrings: []string{`ok`},
-		},
-		{
-			url:                    "/readyz",
-			expectedStatusCode:     http.StatusOK,
-			expectedRespSubStrings: []string{`ok`},
+var defaultHealthCheckConfigs = []healthCheckConfig{
```
Comment (fuweid) [inline tests/e2e/http_health_check_test.go:175]: it seems not related to this change.

**Review by fuweid — APPROVED** [review]
Basically, it looks good.

The pull requests include refactors, for example,

* var => `:=`
* Two lines => one line
* Adding empty line

For reviewer, if non-related change is small, it's easy to be accepted and review
However, there are too many lines changed not related to that `using requires`.
My suggestion is that if linter doesn't blame, just keep it as it is and file other pull request for that.

**Review by ahrtr — APPROVED** [review]
